compiler/src/dotty/tools/dotc/quoted/QuoteDriver.scala

```diff
@@ -2,20 +2,21 @@ package dotty.tools.dotc.quoted
 
 import dotty.tools.dotc.ast.tpd
 import dotty.tools.dotc.Driver
-import dotty.tools.dotc.core.Contexts.Context
+import dotty.tools.dotc.core.Contexts.{Context, ContextBase}
 import dotty.tools.io.{AbstractFile, Directory, PlainDirectory, VirtualDirectory}
 import dotty.tools.repl.AbstractFileClassLoader
 
 import scala.quoted.{Expr, Type}
 import java.net.URLClassLoader
 
-import Toolbox.{Run, Settings, Show}
 import dotty.tools.dotc.tastyreflect.TastyImpl
 
 class QuoteDriver extends Driver {
   import tpd._
 
-  def run[T](expr: Expr[T], settings: Settings[Run]): T = {
+  private[this] val contextBase: ContextBase = new ContextBase
```

Comment:
I don't think this is thread-safe

Reply:
It is not and not intended to be. Each thread should generate its own toolbox. I will add documentation and thick how to check that they do not use the same compiler in two threads.

Reply:
Decided to make it thread-safe by synchronization instead. If someone wants parallelism they should create several instances of the `Toolbox`

Reply:
> Decided to make it thread-safe by synchronization instead.

I don't see any synchronization logic in the code.

Reply:
I must have lost that commit in one of the reading. I will add it again.

Reply:
Searching for `synchronized` still finds nothing here?

Reply:
It is in #4746